### DISCUSSION TITLE
[v6] Benefits of useRoutes Hook

### DISCUSSION THREAD

**Discussion by rwieruch** [remix-run/react-router#8273] [Q&A] [answered]
Is there any benefit of using the `useRoutes` hook over the common way of defining routes? If not, I assume you just left it in there as alternative, because you are using its logic internally anyway and kept the API open.

**Comment by mjackson** [comment, marked as answer]
`useRoutes()` is just a hook that does what `<Routes>` does. We have this API symmetry in a few other places as well, e.g. `useNavigate()` and `<Navigate>`.
